penguin/web/services/session_view.py

```
@@ -48,6 +49,28 @@ def _normalize_non_empty_string(value: Any) -> Optional[str]:
     return normalized or None
 
 
+def _normalize_provider_id(value: Any) -> Optional[str]:
```

Ruff (0.15.15)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)